.github/workflows/shopify-deploy.yml

```diff
@@ -0,0 +1,147 @@
+name: Naturally Sweet Shopify Theme Deployment
+
+on:
+  push:
+    branches:
+      - 'feature/*'  # Trigger for feature branches for theme checks
+      - staging      # Trigger for the staging branch to push as unpublished
+      - main         # Trigger for the main branch to publish the theme
+
+jobs:
+  theme-check:
+    runs-on: ubuntu-latest
+    if: github.ref_name == 'main' || github.ref_name == 'staging' || startsWith(github.ref_name, 'feature/')
+    container:
+      image: aaronmedinaaligent/shopify-cli-base:latest
+      credentials:
+        username: ${{ secrets.DOCKER_HUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_SECRET_KEY }}
+    steps:      
+      - name: Checkout code
+        uses: actions/checkout@v2
```

Comment:
Most GHA default actions are `@v4` now ... otherwise you'll get pipeline warnings about deprecated Node